src/use_file.rs

```diff
@@ -32,23 +36,21 @@ pub fn getrandom_inner(dest: &mut [MaybeUninit<u8>]) -> Result<(), Error> {
 // Returns the file descriptor for the device file used to retrieve random
 // bytes. The file will be opened exactly once. All subsequent calls will
 // return the same file descriptor. This file descriptor is never closed.
-fn get_rng_fd() -> Result<libc::c_int, Error> {
-    static FD: AtomicUsize = AtomicUsize::new(FD_UNINIT);
-
-    fn get_fd() -> Option<libc::c_int> {
+fn get_rng_fd() -> Result<RawFd, Error> {
+    static FD: AtomicI32 = AtomicI32::new(FD_UNINIT);
+    fn get_fd() -> Option<RawFd> {
         match FD.load(Relaxed) {
             FD_UNINIT => None,
-            val => Some(val as libc::c_int),
+            val => Some(val),
         }
     }
 
     #[cold]
-    fn get_fd_locked() -> Result<libc::c_int, Error> {
-        // SAFETY: We use the mutex only in this method, and we always unlock it
-        // before returning, making sure we don't violate the pthread_mutex_t API.
-        static MUTEX: Mutex = Mutex::new();
-        unsafe { MUTEX.lock() };
-        let _guard = DropGuard(|| unsafe { MUTEX.unlock() });
+    fn get_fd_locked() -> Result<RawFd, Error> {
+        static MUTEX: Mutex<()> = Mutex::new(());
```

Comment:
Another possibility, ideally deferred to a future PR as it is riskier, would be to replace the double-checked locking with a simple `RwLock<File>`, on the assumption that acquiring the read lock is fast enough that avoiding doing it isn't necessary.

Reply:
As I mentioned elsewhere, I filed https://github.com/rust-lang/rust/issues/126239 to get libstd to document the widely-assumed "fact" that Mutxes have acquire/release semantis.